fdbserver/DDRelocationQueue.actor.cpp

```diff
@@ -1979,20 +2102,24 @@ ACTOR Future<bool> rebalanceReadLoad(DDQueue* self,
                                      DataMovementReason moveReason,
                                      Reference<IDataDistributionTeam> sourceTeam,
                                      Reference<IDataDistributionTeam> destTeam,
+                                     Optional<Reference<IDataDistributionTeam>> randomTeam,
                                      bool primary,
                                      TraceEvent* traceEvent) {
 	if (g_network->isSimulated() && g_simulator.speedUpSimulation) {
 		traceEvent->detail("CancelingDueToSimulationSpeedup", true);
 		return false;
 	}
 
+	state Future<bool> reduceShards = rebalanceByReduceShard(self, randomTeam, moveReason, primary);
+	wait(ready(reduceShards));
+
 	state std::vector<KeyRange> shards = self->shardsAffectedByTeamFailure->getShardsFor(
 	    ShardsAffectedByTeamFailure::Team(sourceTeam->getServerIDs(), primary));
 	traceEvent->detail("ShardsInSource", shards.size());
 	// For read rebalance if there is just 1 hot shard remained, move this shard to another server won't solve the
 	// problem.
-	// TODO: This situation should be solved by split and merge
-	if (shards.size() <= 1) {
+	// UPDATE: This problem can be solved by dynamic replication & split,merge
+	if (shards.size() < 1) {
```

Comment:
If `DYNAMIC_REPLICATION` is disabled and we have exactly one shard, this change could result in a single shard being moved back and forth between two teams. We would execute all the code up to selectMoveReason and decide to do MOVE_SHARD

Reply:
Fixed, added extra condition here to check Knobs.
`if (shards.size() <= 1 && !SERVER_KNOBS->DYNAMIC_REPLICATION_ENABLED) {`